packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -218,6 +238,60 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
                 return;
               }
 
+              case 'tool-calls-chunk': {
+                if (value.tool_call_delta) {
+                  if (value.tool_call_delta.index != toolCalls.length - 1) {
+                    const toolCallId = generateId();
+
+                    toolCalls.push({
```

Comment:
instead of push, you can set at index. it'll increase the size of the array automatically afaik and guarantee that we have a match of `index` to array position

Reply:
agreed, updated!